File: leetcode/035-Search-Insert-Position/tina.swift
class Solution {
    func searchInsert(_ nums: [Int], _ target: Int) -> Int {
        var low = 0
        var high = nums.count - 1
        var mid = 0
        
        while low <= high {
            mid = low + (high - low) / 2
            
            if nums[mid] == target {
                return mid
            }
            
            if target > nums[mid] {
                low = mid + 1
            } else {
                high = mid - 1
            }
        }
        
        return low
    }
}

File: leetcode/240-Search-a-2D-Matrix-II/tina.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if (matrix.size() == 0 || matrix[0].size() == 0) return false;
        
        int m = matrix.size();
        int n = matrix[0].size();
        
        int x = m - 1;
        int y = 0;
        
        while (x >= 0 && y < n) {
            if (target == matrix[x][y]) {
                return true;
            }
            
            if (target < matrix[x][y]) {
                x--;
            } else {
                y++;
            }
        }
        
        return false;
    }
};
